### PR TITLE
(Support) Add Bookmark

### PR DESCRIPTION
**Pull request by dcupidon**
Exposed an API that programmatically create bookmarks once given a title and page number.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -651,6 +651,22 @@ Returns a Future.
 PdftronFlutter.importBookmarkJson("{\"0\": \"Page 1\", \"3\": \"Page 4\"}");
 ```
 
+#### addBookmark
+Creates a new bookmark given a title and page number.
```
Comment (JamesTron) [inline API.md:655]:
```suggestion
Creates a new bookmark with the given title and page number.
```

File: API.md [hunk 2]
```diff
@@ -651,6 +651,22 @@ Returns a Future.
...
+
+Parameters:
+
+Name | Type | Description
+--- | --- | ---
+title | String | Title of the bookmark
+pageNumber | int | The page number of the new bookmark. It is 1-indexed
```
Comment (JamesTron) [inline API.md:662]:
```suggestion
pageNumber | int | The page number of the new bookmark. It is 1-indexed.
```

File: ios/Classes/PdftronFlutterPlugin.h [hunk 3]
```diff
@@ -202,6 +203,7 @@ static NSString * const PTXfdfCommandArgumentKey = @"xfdfCommand";
 static NSString * const PTXfdfArgumentKey = @"xfdf";
 static NSString * const PTBookmarkJsonArgumentKey = @"bookmarkJson";
 static NSString * const PTPageNumberArgumentKey = @"pageNumber";
+static NSString * const PTTitleArgumentKey = @"title";
```
Comment (JamesTron) [inline ios/Classes/PdftronFlutterPlugin.h:206]:
```suggestion
static NSString * const PTBookmarkTitleArgumentKey = @"bookmarkTitle";
```

**Review by JamesTron — APPROVED** [review]

**Review by sgong-pdftron — APPROVED** [review]

File: ios/Classes/PdftronFlutterPlugin.h (hunk 3 above)
Comment (akwanpdf) [reply ios/Classes/PdftronFlutterPlugin.h]: renamed key, kept string for consistency with Android

**Pull request merged**